tests/Unit/Domain/DomainTestHelpers.cpp

```diff
@@ -0,0 +1,56 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/Domain/DomainTestHelpers.hpp"
+
+#include <catch.hpp>
+#include <typeinfo>
+
+#include "Domain/Block.hpp"
+#include "Domain/BlockNeighbor.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/Domain.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+
+template <size_t VolumeDim>
+void test_domain_construction(
+    const Domain<VolumeDim, Frame::Inertial>& domain,
+    const std::vector<
+        std::unordered_map<Direction<VolumeDim>, BlockNeighbor<VolumeDim>>>&
+        expected_block_neighbors,
+    const std::vector<std::unordered_set<Direction<VolumeDim>>>&
+        expected_external_boundaries,
+    const std::vector<std::unique_ptr<
+        CoordinateMapBase<Frame::Logical, Frame::Inertial, VolumeDim>>>&
+        expected_maps) {
+  const auto& blocks = domain.blocks();
+  CHECK(blocks.size() == expected_external_boundaries.size());
+  CHECK(blocks.size() == expected_block_neighbors.size());
+  CHECK(blocks.size() == expected_maps.size());
+  for (size_t i = 0; i < blocks.size(); ++i) {
+    const auto& block = blocks[i];
+    CHECK(block.id() == i);
+    CHECK(block.neighbors() == expected_block_neighbors[i]);
+    CHECK(block.external_boundaries() == expected_external_boundaries[i]);
+    CHECK(typeid(*expected_maps[i]) == typeid(block.coordinate_map()));
```

Comment:
This only checks the types are as expected, not that the maps are equivalent; Will suggests testing them at a few points in the block; Eamonn suggests modifying the are_maps_equal in TestMapHelpers to do this check.